pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/painless/UpsertPainless.java

```diff
@@ -41,22 +43,33 @@ public void export() {
         updateByQueryRequest.setMaxRetries(5);
         updateByQueryRequest.setRefresh(true);
         updateByQueryRequest.setAbortOnVersionConflict(false);
+
+        Throwable ioException = null;
         for (int tryNum = 1; tryNum < 3; tryNum++) {
-            if (executeExport(clientInstance, updateByQueryRequest, tryNum)) {
+            ioException = executeExport(clientInstance, updateByQueryRequest, tryNum);
+            if (ioException == null) {
```

Comment:
Some observations around this block:
1) line 43 sets a MaxRetries to 5, but this loop seems to be hard coded to 3.  Is that intended?
2) Is it important to know which attempt succeeded?
3) Is the only type of exception thrown an ioException?  If not, the variable name seems misleading.

Reply:
I had no reason to update this code per se, but did so since it was incorrect: if DSM could not connect to ES the code just behaved as if success. One of those hidden errors where ES gets out of sync with DSM DB and we don't know about it until it impacts a customer.
It *might* be important know if it took multiple attempts to connect, but I think that is best left to the engineer who analyzes and fixes our ES connection issues.
In the existing version of this code `executeExport` only catches IOExceptions for the sole purpose of retrying, which is split into two methods. All other exceptions throw out of this method. What I did was made that clearer, save the IOException message, since it likely has important information, and ensure we recored the error. Since I do not have a good way of creating an automated test for this code I wanted to fix the problem but limit the impact of the change.